Remotion/Data/DomainObjects/Persistence/Rdbms/DbCommandBuilders/QueryDbCommandBuilder.cs

```diff
@@ -28,51 +31,67 @@ namespace Remotion.Data.DomainObjects.Persistence.Rdbms.DbCommandBuilders
   /// </summary>
   public class QueryDbCommandBuilder : DbCommandBuilder
   {
-    private readonly string _statement;
-    private readonly QueryParameterWithDataParameterDefinition[] _parametersWithDefinition;
-
     public QueryDbCommandBuilder (string statement, IEnumerable<QueryParameterWithDataParameterDefinition> parameters, ISqlDialect sqlDialect)
         : base(sqlDialect)
     {
       ArgumentUtility.CheckNotNull("statement", statement);
       ArgumentUtility.CheckNotNull("parameters", parameters);
 
-      _statement = statement;
-      _parametersWithDefinition = parameters.ToArray();
+      Statement = statement;
+      ParametersWithDefinition = parameters.ToArray();
     }
 
+    protected string Statement { get; }
+    protected QueryParameterWithDataParameterDefinition[] ParametersWithDefinition { get; }
+
     public override IDbCommand Create (IDbCommandFactory dbCommandFactory)
     {
       ArgumentUtility.CheckNotNull("dbCommandFactory", dbCommandFactory);
 
       var command = dbCommandFactory.CreateDbCommand();
 
-      var statement = _statement;
-      foreach (var parameterWithDefinition in _parametersWithDefinition)
-      {
-        var queryParameterValue = parameterWithDefinition.QueryParameter.Value;
-
-        if (parameterWithDefinition.QueryParameter.ParameterType == QueryParameterType.Text)
-        {
-          Assertion.DebugAssert(
-              queryParameterValue is string,
-              "parameterWithType.QueryParameter.Value is string when parameterWithType.QueryParameter.ParameterType == Text");
-          var queryParameterValueAsString = (string)queryParameterValue;
+      var statementBuilder = new StringBuilder(Statement.Length);
+      var statementIndex = 0;
 
-          statement = statement.Replace(parameterWithDefinition.QueryParameter.Name, queryParameterValueAsString);
```

Comment:
okay, existing solution was a bit...iffy beceause it didn't account for parameter-name overlap. I would say that this isn't a feature but a bug and nobdy ever noticed it or there would be a unit test to confirm this behavior.

Reply:
To make matters worse, for us and our design, there's no naming rule for how the parameter name works with text parameters.

Reply:
Lastly, based on available usage information, we do have one user of the Text parameter, there it is used correctly with `@` prefixed.